lib/Data/Chronicle/Reader.pm

```diff
@@ -105,9 +112,16 @@ sub get {
     my $name     = shift;
 
     my $key         = $category . '::' . $name;
-    my $cached_data = $self->cache_reader->get($key);
 
-    return JSON::from_json($cached_data) if defined $cached_data;
+    if ( defined $self->cache_reader ) {
+        my $cached_data = $self->cache_reader->get($key);
+        return JSON::from_json($cached_data) if defined $cached_data;
+    }
+
+    if ( defined $self->memory_map ) {
+        return $self->memory_map->{$key};
+    }
```

Comment:
I'd suggest to use separate class `MemoryReader` instead of overloading the current class, and, it will work faster, because we don't need to do that checks at runtime.